### DISCUSSION TITLE
Can't sync Rstudio projects

### DISCUSSION THREAD

**Discussion by giabaio** [abraunegg/onedrive#2574] [Q&A] [answered]
I have an issue with using [`Rstudio` projects](https://support.posit.co/hc/en-us/articles/200526207-Using-RStudio-Projects). If a project is stored in a folder that is synced by `onedrive`, there's a [known problem](https://community.rstudio.com/t/rstudio-projects-in-onedrive/153275). So, if the folder in question is `~/my-files/r-project`, `Rstudio` will create a sub-folder `.Rproj-user`, which won't sync because some of the files it contains get locked. 

Is there a way to tell my `.config` file that I want to skip syncing ***all*** folders named `.Rproj-user`? These can be located anywhere under the `OneDrive` folder, so I think it's not practical to define a set path...

I hope this makes sense,
Thanks!
Gianluca

**Comment by abraunegg** [comment, marked as answer]
@giabaio 
> Is there a way to tell my .config file that I want to skip syncing all folders named .Rproj-user? These can be located anywhere under the OneDrive folder, so I think it's not practical to define a set path...

Please read the documentation as this sort of capability is explained in detail:

**v2.4.x Documentation**
* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#how-to-skip-directories-from-syncing

**v2.5.x Documentation**
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-alpha-4/docs/usage.md#how-to-skip-directories-from-syncing
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-alpha-4/docs/application-config-options.md#skip_dir

**Comment by giabaio** [comment]
Thank you. I had looked at the documentation, but haven't quite appreciated it, especially the newest version. I suppose I was looking for a shortcut --- and I guess `skip_dotfiles = "true"` is probably the best way around this for me... I'll think about whether this may cause some unintended issues, but thanks for pointing me out to this!

**Reply by abraunegg** [reply]
> and I guess skip_dotfiles = "true" is probably the best way around this for me

'skip_dir' is the correct solution here, which is why I pointed you at the 'skip_dir' documentation.

**Comment by giabaio** [comment]
Thank you --- perhaps I haven't explained my issue clearly (or maybe I am a bit thick and don't get your pointers)... Sorry, either way.

I know how `skip_dir` works and I understand that it *is* the best way to skip a specific folder from syncing. I have a bunch of folders (all named `.Rproj_user`, but all with different content) that are scattered all over my `~/` folder. Whenever I create a new `R` project, a new `.Rproj_user` folder will be created, which will have issues syncing with OneDrive (this is *not* a problem of the Linux client). 

I can manually add the path to each `R` project whenever I set a new one up, but I wanted to avoid this, because, potentially, I have many of them. I was wondering whether there was a more general way --- again, I know I could use wildcards and, perhaps, I could do something like
```
skip_dir="*/.Rproj*|**/.Rproj*|***/.Rproj*"
```
or something...

Does this make more sense?

**Reply by abraunegg** [reply]
I understand what you are wanting to do.

Your solution is, as per the documentation:

```
skip_dir = ".Rproj_user"
```

This will skip all instances, wherever it is located.

**Comment by giabaio** [comment]
Ah. I had not appreciated this. Thank you,

---

Gianluca Baio
Professor of Statistics and Health Economics | Head of Department


Department of Statistical Science | University College London
1-19 Torrington Place, London, WC1E 6BT, UK
Twitter: @stats_ucl<https://twitter.com/stats_UCL> | Instagram: @stats_UCL<https://instagram.com/ucl.stats/>


Telephone: +44(0)20 7679 1248 (internal: 41248)
Website: http://www.homepages.ucl.ac.uk/~ucakgba
Blog: https://gianluca.statistica.it/blog/
Book a meeting with me: https://book.morgen.so/gianlucabaio/book-me

________________________________
